### PR TITLE
Remove long disabled TR_ASSERTC macros

### PR DESCRIPTION
**Pull request by 0xdaryl**
The preprocessor macro does not even exist any longer.

Signed-off-by: Daryl Maier <maier@ca.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins build all

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -12507,8 +12502,6 @@ OMR::Z::TreeEvaluator::arraysetEvaluator(TR::Node * node, TR::CodeGenerator * cg
    //     baseReg = cg->gprClobberEvaluate(baseAddr);
 
    cg->decReferenceCount(elemsExpr);
-   //TR_ASSERTC( elemsExpr->getSize() <=4,comp, "If this is not true we should not be truncating");
-   //someone should investigate why this assertion is not being hit
```
Comment (dsouzai) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:12511]: 😂

**Review by dsouzai — APPROVED** [review]
LGTM

**Comment by dsouzai** [comment]
jenkins build win

**Pull request merged**